src/Symfony/Component/EventDispatcher/EventDispatcher.php

```diff
@@ -242,7 +243,8 @@ protected function callListeners(iterable $listeners, string $eventName, $event)
             if ($stoppable && $event->isPropagationStopped()) {
                 break;
             }
-            $listener($event instanceof Event ? $event : new WrappedEvent($event), $eventName, $this);
+            // @deprecated: the ternary operator is part of a BC layer and should be removed in 5.0
+            $listener($event instanceof Event || !$listener instanceof WrappedListener ? $event : new WrappedEvent($event), $eventName, $this);
```

Comment:
`$event instanceof Event` is always false because of the early return l234 
and `!$listener instanceof WrappedListener` is always true as listeners are optimized at this point (plain `\Closure`). 
Broken test case:
```
diff --git a/src/Symfony/Component/EventDispatcher/Tests/Debug/TraceableEventDispatcherTest.php b/src/Symfony/Component/EventDispatcher/Tests/Debug/TraceableEventDispatcherTest.php
index 8fd0305b0e..db3176dcd1 100644
--- a/src/Symfony/Component/EventDispatcher/Tests/Debug/TraceableEventDispatcherTest.php
+++ b/src/Symfony/Component/EventDispatcher/Tests/Debug/TraceableEventDispatcherTest.php
@@ -18,6 +18,7 @@ use Symfony\Component\EventDispatcher\EventDispatcher;
 use Symfony\Component\EventDispatcher\EventDispatcherInterface;
 use Symfony\Component\EventDispatcher\EventSubscriberInterface;
 use Symfony\Component\Stopwatch\Stopwatch;
+use Symfony\Contracts\EventDispatcher\Event as ContractsEvent;
 
 class TraceableEventDispatcherTest extends TestCase
 {
@@ -151,6 +152,15 @@ class TraceableEventDispatcherTest extends TestCase
         $this->assertArrayHasKey('stub', $listeners[0]);
     }
 
+    public function testDispatchContractsEvent()
+    {
+        $tdispatcher = new TraceableEventDispatcher(new EventDispatcher(), new Stopwatch());
+        $tdispatcher->addListener('foo', function () {}, 5);
+        $tdispatcher->dispatch(new ContractsEvent(), 'foo');
+        $listeners = $tdispatcher->getCalledListeners();
+        $this->assertArrayHasKey('stub', $listeners[0]);
+    }
+
     public function testGetCalledListenersNested()
     {
         $tdispatcher = null;
```